front/src/common/IntervalsDataViz/Scales.tsx

```diff
@@ -1,54 +1,123 @@
+    const graphWidth = wrapper.offsetWidth;
+    const leftPadding = wrapper.offsetLeft;
+
+    // compute the number of ticks, knowing that space between 2 ticks should be 200px
+    const ticksCount = Math.round(graphWidth / 200);
+
+    // compute the default step between 2 values (we want to display only value in km)
+    const step = roundNumber((_end - _begin) / ticksCount / 1000) * 1000;
+
+    // compute the ticks (value and position)
+    const newtTicks = [{ value: _begin, position: leftPadding, isExtremity: true }];
+    if (step > 0) {
+      // find the first tick value (first integer above begin (in km) or begin if it is x km)
+      let index = isRoundKm(begin) ? 1 : 0;
+      const firstTick = isRoundKm(begin) ? _begin : roundNumber(begin / 1000, true) * 1000;
 
-  const inf = roundNumber(begin, true);
-  const sup = roundNumber(end, false);
-  const step = roundNumber((end - begin) / ticksCount / 2, true);
+      // check the second tick is not too close to the first one
+      const secondTickPosition = getPositionFromValue(
+        step * index + firstTick,
+        _begin,
+        _end,
+        graphWidth,
+        leftPadding
+      );
+      if (secondTickPosition - leftPadding < 20) {
+        index += 1;
+      }
 
-  /** redraw the scale when window resized horizontally */
+      // compute the interior ticks
+      while (index * step + firstTick < _end) {
+        newtTicks.push({
+          value: step * index + firstTick,
+          position: getPositionFromValue(
+            step * index + firstTick,
+            _begin,
+            _end,
+            graphWidth,
+            leftPadding
+          ),
+          isExtremity: false,
+        });
+        index += 1;
+      }
+
+      // check before last tick is not too close to the last tick
+      if (newtTicks[newtTicks.length - 1].position - leftPadding > graphWidth - 20) {
+        newtTicks.pop();
+      }
+    }
+    newtTicks.push({ value: _end, position: graphWidth + leftPadding, isExtremity: true });
+    return newtTicks;
+  };
+
+  useEffect(() => {
+    const newTicks = computeTicks(begin, end, wrapper);
+    setTicks(newTicks);
+  }, [begin, end, wrapper]);
+
+  /**
+   * When the window is resized horizontally
+   * => we recompute the operationalPoints4viz
+   */
   useEffect(() => {
     const debounceResize = () => {
       let debounceTimeoutId;
       clearTimeout(debounceTimeoutId);
       debounceTimeoutId = setTimeout(() => {
-        const graphWidth = document.getElementById('linear-metadata-dataviz-content')?.offsetWidth;
-        if (graphWidth) {
-          setTicksCount(Math.round(graphWidth / 100));
-        }
+        const newTicks = computeTicks(begin, end, wrapper);
+        setTicks(newTicks);
       }, 15);
     };
     window.addEventListener('resize', debounceResize);
     return () => {
       window.removeEventListener('resize', debounceResize);
     };
-  }, []);
```

Comment:
why setTicks ?

Reply:
I removed it, it was not useful, I forgot to remove it before pushing :) thanks !